Content.Shared/Weapons/Melee/Components/InfectOnMeleeComponent.cs

```diff
@@ -0,0 +1,27 @@
+using Robust.Shared.Audio;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Weapons.Melee;
+
+[RegisterComponent]
+[NetworkedComponent]
+public sealed partial class InfectOnMeleeComponent : Component
+{
+    /// <summary>
+    /// infection chance determines the % chance that target will be infected.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    public float InfectionChance = 0.15f;
+
+    /// <summary>
+    /// If this is true then target will be cluwned.
+    /// </summary>
+    [DataField("cluwnification")]
+    public bool Cluwinification = false;
```

Comment:
1. Infection needs to be a datafield and networked.
2. This should not be tied to cluwne and zombies needs to use it.